lib/src/data/datasource/mappers/genre_mapper.dart

```diff
@@ -0,0 +1,11 @@
+import '../../../domain/entity/genre_entity.dart';
+import '../../model/genre_data.dart';
+
+class GenreMapper {
+  static Genre fromDataToEntity(GenreData genre) {
+    return Genre(
+      id: genre.id,
+      name: genre.name,
+    );
+  }
```

Comment:
You can define a Genre Entity like this on domain layer
```
class GenreEntity {
  GenreEntity{
    required this.id,
    required this.name
  });

  final int id;
  final String name;
}
```

Reply:
on data layer you are going to have a GenreModel like this
```
class GenreModel extends GenreEntity {
  GenreModel({
    required int id,
    required String name,
  }) : super(
          id: id,
          name: name,
        );

  factory GenreModel.fromJson(Map<String, dynamic> json) => GenreModel(...);
}
```

Reply:
If you notice, on domain we define the structure that we need to use and presentation is going to use, but on data layer we know how to build it